--- Makefile.orig	2009-03-04 20:35:58.000000000 -0300
+++ Makefile	2009-03-04 20:36:32.000000000 -0300
@@ -23,7 +23,7 @@
 # Where figlet will search first for fonts (the ".flf" files).
 DEFAULTFONTDIR = /usr/local/share/figlet
 # Use this definition if you can't put things in /usr/local/share/figlet
-DEFAULTFONTDIR = fonts
+#DEFAULTFONTDIR = fonts
 
 # The filename of the font to be used if no other is specified
 #   (standard.flf is recommended, but any other can be used).
