--- setup.py.orig	2011-11-27 13:48:21.000000000 -0200
+++ setup.py	2011-11-27 13:48:23.000000000 -0200
@@ -146,6 +146,9 @@
             # darwin ports installation directories
             add_directory(library_dirs, "/opt/local/lib")
             add_directory(include_dirs, "/opt/local/include")
+            # darwin (OSX 10.7)
+            add_directory(library_dirs, "/usr/X11/lib")
+            add_directory(include_dirs, "/usr/X11/include")
 
         add_directory(library_dirs, "/usr/local/lib")
         # FIXME: check /opt/stuff directories here?
