--- Makefile.in.orig	2010-03-02 19:45:05.000000000 -0300
+++ Makefile.in	2010-03-02 19:45:11.000000000 -0300
@@ -38,8 +38,8 @@
 @HAVE_LIBLTDL_TRUE@am__append_1 = -DPKGLIBDIR="\"$(pkglibdir)\""
 @HAVE_DISTRO_TRUE@am__append_2 = -DDISTRO="\"@DISTRO@\""
 bin_PROGRAMS = sox$(EXEEXT)
-EXTRA_PROGRAMS = example0$(EXEEXT) example1$(EXEEXT) example2$(EXEEXT) \
-	example3$(EXEEXT) sox_sample_test$(EXEEXT)
+#EXTRA_PROGRAMS = example0$(EXEEXT) example1$(EXEEXT) example2$(EXEEXT) \
+#	example3$(EXEEXT) sox_sample_test$(EXEEXT)
 @HAVE_PNG_TRUE@am__append_3 = spectrogram.c
 @HAVE_MAGIC_TRUE@am__append_4 = @MAGIC_LIBS@
 @HAVE_OPENMP_TRUE@am__append_5 = @GOMP_LIBS@
