diff --git a/Makefile b/Makefile
index e6e4987..cfd5fe0 100644
--- a/Makefile
+++ b/Makefile
@@ -427,7 +427,7 @@ osx:	osxok an
 	SPECIALS="SSLINCLUDE=/usr/include/openssl SSLLIB=/usr/lib SSLCERTS=/System/Library/OpenSSL/certs SSLKEYS=/System/Library/OpenSSL/private GSSINCLUDE=/usr/include GSSLIB=/usr/lib"
 
 osxok:
-	@$(SH) -c '(test ! -f /usr/include/pam/pam_appl.h ) || make osxwarn'
+	@$(SH) -c '(test ! -f /usr/include/security/pam_appl.h ) || make osxwarn'
 	@$(TOUCH) osxok
 
 osxwarn:
diff --git a/src/osdep/unix/ckp_pam.c b/src/osdep/unix/ckp_pam.c
index 60c6c1f..674941c 100644
--- a/src/osdep/unix/ckp_pam.c
+++ b/src/osdep/unix/ckp_pam.c
@@ -27,11 +27,7 @@
  */
 
 
-#ifdef MAC_OSX_KLUDGE		/* why can't Apple be compatible? */
-#include <pam/pam_appl.h>
-#else
 #include <security/pam_appl.h>
-#endif
 
 struct checkpw_cred {
   char *uname;			/* user name */
