--- Makefile.in.orig	2008-12-06 17:51:59.000000000 -0200
+++ Makefile.in	2008-12-06 17:54:46.000000000 -0200
@@ -66,7 +66,7 @@
 ERLANG_ILIBDIR    = @libdir@/erlang
 
 # You can *not* change these two, they have to stay this way for now.
-ERLANG_BINDIR     = $(ERLANG_LIBDIR)/bin
+ERLANG_BINDIR     = $(ERLANG_ILIBDIR)/bin
 ERLANG_ERTSBINDIR = $(ERLANG_LIBDIR)/$(ERTS)/bin
 
 # The directory in which man pages for above executables are put
