--- configure.orig	Sat Mar 25 00:31:41 2006
+++ configure	Sat Mar 25 00:36:25 2006
@@ -11265,14 +11265,10 @@
       esac
     done ;;
     gettext-fix )
-  sed -e '/^mkinstalldirs *=/a\' \
-      -e "install_sh=$install_sh" \
-      -e 's/^mkinstalldirs *=.*/mkinstalldirs=$(MKINSTALLDIRS)/' \
+  sed -e 's/^mkinstalldirs *=.*/mkinstalldirs=$(MKINSTALLDIRS)/' \
       intl/Makefile > intl/Makefile.tmp
   mv intl/Makefile.tmp intl/Makefile
-  sed -e '/^mkinstalldirs *=/a\' \
-      -e "install_sh=$install_sh" \
-      -e 's/^mkinstalldirs *=.*/mkinstalldirs=$(MKINSTALLDIRS)/' \
+  sed -e 's/^mkinstalldirs *=.*/mkinstalldirs=$(MKINSTALLDIRS)/' \
       po/Makefile > po/Makefile.tmp
   mv po/Makefile.tmp po/Makefile ;;
   esac
