--- config.layout.orig	2007-12-01 16:53:37.000000000 -0200
+++ config.layout	2007-12-01 16:57:08.000000000 -0200
@@ -9,6 +9,30 @@
 ##    (This may become a configurable parameter at some point.)
 ##
 
+#   Rudix 2007 layout
+<Layout Rudix>
+    prefix:        /usr/local
+    exec_prefix:   ${prefix}
+    bindir:        ${prefix}/bin
+    sbindir:       ${prefix}/sbin
+    libdir:        ${prefix}/lib
+    libexecdir:    ${prefix}/libexec/httpd
+    mandir:        ${prefix}/share/man
+    sysconfdir:    ${prefix}/etc/httpd
+    datadir:       ${prefix}/var/www
+    installbuilddir: ${datadir}/build
+    errordir:      ${datadir}/error
+    iconsdir:      ${datadir}/icons
+    htdocsdir:     ${datadir}/html
+    manualdir:     ${datadir}/manual
+    cgidir:        ${datadir}/cgi-bin
+    includedir:    ${prefix}/include/httpd
+    localstatedir: ${prefix}/var
+    runtimedir:    ${localstatedir}/run
+    logfiledir:    ${localstatedir}/log/httpd
+    proxycachedir: ${localstatedir}/cache/httpd
+</Layout>     
+
 #   Classical Apache path layout.
 <Layout Apache>
     prefix:        /usr/local/apache2
