--- offlineimap/__init__.py.orig	2014-10-28 13:57:02.000000000 -0200
+++ offlineimap/__init__.py	2014-10-28 13:57:12.000000000 -0200
@@ -1,7 +1,7 @@
 __all__ = ['OfflineImap']
 
 __productname__ = 'OfflineIMAP'
-__version__     = "6.5.5"
+__version__     = "6.5.6"
 __copyright__   = "Copyright 2002-2013 John Goerzen & contributors"
 __author__      = "John Goerzen"
 __author_email__= "john@complete.org"
