--- sha1.c.orig	2009-03-11 21:36:40.000000000 -0300
+++ sha1.c	2009-03-11 21:37:28.000000000 -0300
@@ -159,7 +159,7 @@
 SHA1Pad(SHA1_CTX *context)
 {
 	uint8_t finalcount[8];
-	uint i;
+	uint32_t i;
 
 	for (i = 0; i < 8; i++) {
 		finalcount[i] = (uint8_t)((context->count >>
@@ -174,7 +174,7 @@
 void
 SHA1Final(uint8_t digest[SHA1_DIGEST_LENGTH], SHA1_CTX *context)
 {
-	uint i;
+	uint32_t i;
 
 	SHA1Pad(context);
 	if (digest) {
