--- libguile/Makefile.in.orig	2008-12-08 15:49:23.000000000 -0200
+++ libguile/Makefile.in	2008-12-16 22:44:34.000000000 -0200
@@ -493,7 +493,7 @@
     c-tokenize.lex version.h.in						\
     scmconfig.h.top gettext.h measure-hwm.scm
 
-snarfcppopts = $(DEFS) $(AM_CPPFLAGS) $(CPPFLAGS) $(CFLAGS)
+snarfcppopts = $(DEFS) $(AM_CPPFLAGS) $(CPPFLAGS)
 SUFFIXES = .x .doc
 preinstguile = $(top_builddir_absolute)/pre-inst-guile
 preinstguiletool = GUILE="$(preinstguile)" $(top_srcdir)/scripts
