--- guile-readline/Makefile.in.orig	2008-12-16 23:03:33.000000000 -0200
+++ guile-readline/Makefile.in	2008-12-16 23:02:31.000000000 -0200
@@ -235,7 +235,7 @@
 libguilereadline_v_@LIBGUILEREADLINE_MAJOR@_la_LDFLAGS = -version-info @LIBGUILEREADLINE_INTERFACE@ -export-dynamic -no-undefined
 BUILT_SOURCES = readline.x
 pkginclude_HEADERS = readline.h
-snarfcppopts = $(DEFS) $(INCLUDES) $(CPPFLAGS) $(CFLAGS)
+snarfcppopts = $(DEFS) $(INCLUDES) $(CPPFLAGS)
 SUFFIXES = .x
 EXTRA_DIST = LIBGUILEREADLINE-VERSION ChangeLog-2008
 MKDEP = gcc -M -MG $(DEFS) $(INCLUDES) $(CPPFLAGS) $(CFLAGS)
