--- Makefile.orig	2012-06-16 18:17:24.000000000 -0300
+++ Makefile	2012-06-16 18:18:16.000000000 -0300
@@ -281,7 +281,7 @@
 	cd $(LIBDIR); rm -f dllbigarray.so dlllabltk.so dllnums.so \
 	  dllthreads.so dllunix.so dllgraphics.so dllmldbm.so dllstr.so \
 	  dlltkanim.so
-	cd byterun; $(MAKE) install
+	cd byterun; $(MAKE) BINDIR=$(BINDIR) install
 	cp ocamlc $(BINDIR)/ocamlc$(EXE)
 	cp ocaml $(BINDIR)/ocaml$(EXE)
 	cd stdlib; $(MAKE) install
