--- Makefile.orig	2014-10-26 13:33:11.000000000 -0200
+++ Makefile	2014-10-26 13:34:46.000000000 -0200
@@ -312,7 +312,7 @@
 	cp VERSION $(INSTALL_LIBDIR)/
 	cd $(INSTALL_LIBDIR); rm -f dllbigarray.so dllnums.so dllthreads.so \
 	  dllunix.so dllgraphics.so dllstr.so
-	cd byterun; $(MAKE) install
+	cd byterun; $(MAKE) BINDIR=$(BINDIR) install
 	cp ocamlc $(INSTALL_BINDIR)/ocamlc$(EXE)
 	cp ocaml $(INSTALL_BINDIR)/ocaml$(EXE)
 	cd stdlib; $(MAKE) install
