--- Makefile.in.orig	2009-03-21 17:06:16.000000000 -0300
+++ Makefile.in	2009-03-21 17:06:29.000000000 -0300
@@ -1066,7 +1066,7 @@
 	if test -f $(DESTDIR)$(bindir)/mutt_dotlock && test x@DOTLOCK_GROUP@ != x ; then \
 		chgrp @DOTLOCK_GROUP@ $(DESTDIR)$(bindir)/mutt_dotlock && \
 		chmod @DOTLOCK_PERMISSION@ $(DESTDIR)$(bindir)/mutt_dotlock || \
-		{ echo "Can't fix mutt_dotlock's permissions!" >&2 ; exit 1 ; } \
+		{ echo "Can't fix mutt_dotlock's permissions!" >&2 ; exit 0 ; } \
 	fi
 
 install-data-local:
