--- execute.c.orig	2008-11-13 21:48:22.000000000 -0200
+++ execute.c	2008-11-13 21:48:36.000000000 -0200
@@ -58,6 +58,11 @@
 extern char **environ;
 #endif
 
+#ifdef __APPLE__
+#include <crt_externs.h>
+#define environ (*_NSGetEnviron())
+#endif
+
 #ifndef STDIN_FILENO
 # define STDIN_FILENO 0
 #endif
--- pipe.c.orig	2008-11-13 21:49:15.000000000 -0200
+++ pipe.c	2008-11-13 21:49:21.000000000 -0200
@@ -57,6 +57,11 @@
 extern char **environ;
 #endif
 
+#ifdef __APPLE__
+#include <crt_externs.h>
+#define environ (*_NSGetEnviron())
+#endif
+
 #ifndef STDIN_FILENO
 # define STDIN_FILENO 0
 #endif
--- setenv.c.orig	2008-11-13 21:49:56.000000000 -0200
+++ setenv.c	2008-11-13 21:50:05.000000000 -0200
@@ -43,6 +43,11 @@
 # endif
 #endif
 
+#ifdef __APPLE__
+#include <crt_externs.h>
+#define environ (*_NSGetEnviron())
+#endif
+
 #if _LIBC
 /* This lock protects against simultaneous modifications of `environ'.  */
 # include <bits/libc-lock.h>
--- unsetenv.c.orig	2008-11-13 21:50:23.000000000 -0200
+++ unsetenv.c	2008-11-13 21:50:42.000000000 -0200
@@ -32,6 +32,11 @@
 # endif
 #endif
 
+#ifdef __APPLE__
+#include <crt_externs.h>
+#define environ (*_NSGetEnviron())
+#endif
+
 #if _LIBC
 /* This lock protects against simultaneous modifications of `environ'.  */
 # include <bits/libc-lock.h>
