--- argp-pv.c.orig	2010-02-28 15:55:37.000000000 -0300
+++ argp-pv.c	2010-02-28 15:55:57.000000000 -0300
@@ -20,4 +20,4 @@
    --version is added (unless the ARGP_NO_HELP flag is used), which will
    print this string followed by a newline and exit (unless the
    ARGP_NO_EXIT flag is used).  Overridden by ARGP_PROGRAM_VERSION_HOOK.  */
-const char *argp_program_version;
+const char *argp_program_version = 0;
