diff -r -c iputils-s20151218/Makefile iputils-build/Makefile
*** iputils-s20151218/Makefile	2015-12-17 16:45:51.000000000 +0100
--- iputils-build/Makefile	2019-07-19 10:43:29.011732316 +0200
***************
*** 114,120 ****
  LDLIBS=$(LDLIB) $(ADDLIB)
  
  UNAME_N:=$(shell uname -n)
! LASTTAG:=$(shell git describe HEAD | sed -e 's/-.*//')
  TODAY=$(shell date +%Y/%m/%d)
  DATE=$(shell date --date $(TODAY) +%Y%m%d)
  TAG:=$(shell date --date=$(TODAY) +s%Y%m%d)
--- 114,120 ----
  LDLIBS=$(LDLIB) $(ADDLIB)
  
  UNAME_N:=$(shell uname -n)
! # LASTTAG:=$(shell git describe HEAD | sed -e 's/-.*//')
  TODAY=$(shell date +%Y/%m/%d)
  DATE=$(shell date --date $(TODAY) +%Y%m%d)
  TAG:=$(shell date --date=$(TODAY) +s%Y%m%d)
