--- stackvma-mach.c.orig	2010-02-14 20:34:05.000000000 -0200
+++ stackvma-mach.c	2010-02-14 20:37:32.000000000 -0200
@@ -53,11 +53,11 @@
   for (address = VM_MIN_ADDRESS; more; address += size)
     {
 #ifdef VM_REGION_BASIC_INFO
-      more = (vm_region (task, &address, &size, VM_REGION_BASIC_INFO,
+      more = (mach_vm_region (task, &address, &size, VM_REGION_BASIC_INFO,
                          (vm_region_info_t)&info, &info_count, &object_name)
               == KERN_SUCCESS);
 #else
-      more = (vm_region (task, &address, &size, &protection, &max_protection,
+      more = (mach_vm_region (task, &address, &size, &protection, &max_protection,
                          &inheritance, &shared, &object_name, &offset)
               == KERN_SUCCESS);
 #endif
