--- configure.orig	2004-02-05 07:34:07.000000000 -0200
+++ configure	2008-03-16 19:37:31.000000000 -0300
@@ -504,8 +504,8 @@
 libdir='${exec_prefix}/lib'
 includedir='${prefix}/include'
 oldincludedir='/usr/include'
-infodir='${prefix}/info'
-mandir='${prefix}/man'
+infodir='${prefix}/share/info'
+mandir='${prefix}/share/man'
 
 ac_prev=
 for ac_option
@@ -7098,11 +7098,11 @@
 	;;
       *) # Darwin 1.3 on
       if test -z ${MACOSX_DEPLOYMENT_TARGET} ; then
-      	allow_undefined_flag='-flat_namespace -undefined suppress'
+      	allow_undefined_flag="$LDFLAGS -flat_namespace -undefined suppress"
       else
         case ${MACOSX_DEPLOYMENT_TARGET} in
           10.[012])
-            allow_undefined_flag='-flat_namespace -undefined suppress'
+            allow_undefined_flag="$LDFLAGS -flat_namespace -undefined suppress"
             ;;
           10.*)
             allow_undefined_flag='-undefined dynamic_lookup'
@@ -8974,11 +8974,11 @@
       ;;
     *) # Darwin 1.3 on
       if test -z ${MACOSX_DEPLOYMENT_TARGET} ; then
-      	allow_undefined_flag='-flat_namespace -undefined suppress'
+      	allow_undefined_flag="$LDFLAGS -flat_namespace -undefined suppress"
       else
         case ${MACOSX_DEPLOYMENT_TARGET} in
           10.[012])
-            allow_undefined_flag='-flat_namespace -undefined suppress'
+            allow_undefined_flag="$LDFLAGS -flat_namespace -undefined suppress"
             ;;
           10.*)
             allow_undefined_flag='-undefined dynamic_lookup'
@@ -10052,11 +10052,11 @@
       ;;
     *) # Darwin 1.3 on
       if test -z ${MACOSX_DEPLOYMENT_TARGET} ; then
-      	allow_undefined_flag_CXX='-flat_namespace -undefined suppress'
+      	allow_undefined_flag_CXX="$LDFLAGS -flat_namespace -undefined suppress"
       else
         case ${MACOSX_DEPLOYMENT_TARGET} in
           10.[012])
-            allow_undefined_flag_CXX='-flat_namespace -undefined suppress'
+            allow_undefined_flag_CXX="$LDFLAGS -flat_namespace -undefined suppress"
             ;;
           10.*)
             allow_undefined_flag_CXX='-undefined dynamic_lookup'
@@ -13927,11 +13927,11 @@
 	;;
       *) # Darwin 1.3 on
       if test -z ${MACOSX_DEPLOYMENT_TARGET} ; then
-      	allow_undefined_flag_F77='-flat_namespace -undefined suppress'
+      	allow_undefined_flag_F77="$LDFLAGS -flat_namespace -undefined suppress"
       else
         case ${MACOSX_DEPLOYMENT_TARGET} in
           10.[012])
-            allow_undefined_flag_F77='-flat_namespace -undefined suppress'
+            allow_undefined_flag_F77="$LDFLAGS -flat_namespace -undefined suppress"
             ;;
           10.*)
             allow_undefined_flag_F77='-undefined dynamic_lookup'
@@ -16272,11 +16272,11 @@
 	;;
       *) # Darwin 1.3 on
       if test -z ${MACOSX_DEPLOYMENT_TARGET} ; then
-      	allow_undefined_flag_GCJ='-flat_namespace -undefined suppress'
+      	allow_undefined_flag_GCJ="$LDFLAGS -flat_namespace -undefined suppress"
       else
         case ${MACOSX_DEPLOYMENT_TARGET} in
           10.[012])
-            allow_undefined_flag_GCJ='-flat_namespace -undefined suppress'
+            allow_undefined_flag_GCJ="$LDFLAGS -flat_namespace -undefined suppress"
             ;;
           10.*)
             allow_undefined_flag_GCJ='-undefined dynamic_lookup'
@@ -19086,8 +19086,7 @@
     if test -z "$arch"
     then
 	case "$host" in
-	    i386-*)           ;;
-	    i?86-*)           arch="-march=i486" ;;
+	    i?86-*)           ;;
 	    arm*-empeg-*)     arch="-march=armv4 -mtune=strongarm1100" ;;
 	    armv4*-*)         arch="-march=armv4 -mtune=strongarm" ;;
 	    powerpc-*)        ;;
