--- archive.c.orig	2008-11-29 12:12:58.000000000 -0200
+++ archive.c	2008-11-29 12:14:25.000000000 -0200
@@ -79,6 +79,11 @@
 #define LONG_MIN INT32_MIN
 #endif
 
+/* http://code.google.com/p/xar/issues/detail?id=51 */
+#if LIBXML_VERSION < 20618
+#define xmlDictCleanup()        /* function doesn't exist in older API */
+#endif
+
 static int32_t xar_unserialize(xar_t x);
 void xar_serialize(xar_t x, const char *file);
 
