--- install.sh.orig	Tue Apr 26 15:15:53 2005
+++ install.sh	Sat Sep 17 21:25:13 2005
@@ -7,9 +7,11 @@
 DEST_BIN=${DEST_HOME}/bin
 DEST_SHARE=${DEST_HOME}/lib/p7zip
 DEST_MAN=${DEST_HOME}/man
+DEST_SHARE_FINAL=${DEST_HOME}/lib/p7zip
 [ "$1" ] && DEST_BIN=$1
 [ "$2" ] && DEST_SHARE=$2
 [ "$3" ] && DEST_MAN=$3
+[ "$4" ] && DEST_SHARE_FINAL=$4
 
 use_share="n"
 
@@ -44,7 +46,7 @@
     strip     ${DEST_SHARE}/7za
     chmod 555 ${DEST_SHARE}/7za
     echo "#! /bin/sh" > ${DEST_BIN}/7za
-    echo "${DEST_SHARE}/7za \"\$@\"" >> ${DEST_BIN}/7za
+    echo "${DEST_SHARE_FINAL}/7za \"\$@\"" >> ${DEST_BIN}/7za
     chmod 555 ${DEST_BIN}/7za
   fi
 
@@ -67,7 +69,7 @@
     cp -r bin/Codecs bin/Formats ${DEST_SHARE}/
     chmod 555 ${DEST_SHARE}/*/*
     echo "#! /bin/sh" > ${DEST_BIN}/7z
-    echo "${DEST_SHARE}/7z \"\$@\"" >> ${DEST_BIN}/7z
+    echo "${DEST_SHARE_FINAL}/7z \"\$@\"" >> ${DEST_BIN}/7z
     chmod 555 ${DEST_BIN}/7z
   fi
 
