--- Makefile.in.orig	Fri Mar 31 22:43:52 2006
+++ Makefile.in	Fri Mar 31 22:44:14 2006
@@ -379,12 +379,12 @@
 
 egrep:
 	echo '#!/bin/sh' >$@
-	echo 'exec grep -E $${1+"$$@"}' >>$@
+	echo 'exec gnugrep -E $${1+"$$@"}' >>$@
 	chmod a+x $@
 
 fgrep:
 	echo '#!/bin/sh' >$@
-	echo 'exec grep -F $${1+"$$@"}' >>$@
+	echo 'exec gnugrep -F $${1+"$$@"}' >>$@
 	chmod a+x $@
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
