--- configure.orig	2011-05-08 06:49:33.000000000 -0300
+++ configure	2011-08-27 21:08:41.000000000 -0300
@@ -6825,8 +6825,6 @@
 	extra_functions="$extra_functions fat fat_entry"
 	path="x86/fat x86"
 	fat_path="x86 x86/fat x86/i486
-		  x86/k6 x86/k6/mmx x86/k6/k62mmx
-		  x86/k7 x86/k7/mmx
 		  x86/pentium x86/pentium/mmx
 		  x86/p6 x86/p6/mmx x86/p6/p3mmx x86/p6/sse2
 		  x86/pentium4 x86/pentium4/mmx x86/pentium4/sse2"
