--- coders/png.c.orig	2012-06-12 19:19:21.000000000 -0300
+++ coders/png.c	2012-06-12 20:12:50.000000000 -0300
@@ -2247,11 +2247,6 @@
   png_set_read_user_chunk_fn(ping, image, read_vpag_chunk_callback);
 #endif
 
-#ifdef PNG_READ_CHECK_FOR_INVALID_INDEX_SUPPORTED
-    /* Disable new libpng-1.5.10 feature */
-    png_set_check_for_invalid_index (ping, 0);
-#endif
-
 #if (PNG_LIBPNG_VER < 10400)
 #  if defined(PNG_USE_PNGGCCRD) && defined(PNG_ASSEMBLER_CODE_SUPPORTED) && \
    (PNG_LIBPNG_VER >= 10200) && (PNG_LIBPNG_VER < 10220) && defined(__i386__)
@@ -8963,12 +8958,6 @@
   if (mng_info->write_mng)
     {
       (void) png_permit_mng_features(ping,PNG_ALL_MNG_FEATURES);
-# ifdef PNG_WRITE_CHECK_FOR_INVALID_INDEX_SUPPORTED
-      /* Disable new libpng-1.5.10 feature when writing a MNG because
-       * zero-length PLTE is OK
-       */
-      png_set_check_for_invalid_index (ping, 0);
-# endif
     }
 
 #else
