--- hoc.y.orig	2008-12-13 14:16:48.000000000 -0200
+++ hoc.y	2008-12-13 14:17:17.000000000 -0200
@@ -3,6 +3,7 @@
 #include "hoc.h"
 #define	code2(c1,c2)	code(c1); code(c2)
 #define	code3(c1,c2,c3)	code(c1); code(c2); code(c3)
+  extern int indef;
 %}
 %union {
 	Symbol	*sym;	/* symbol table pointer */
@@ -309,10 +310,6 @@
 main(int argc, char* argv[])	/* hoc6 */
 {
 	static int first = 1;
-#ifdef YYDEBUG
-	extern int yydebug;
-	yydebug=3;
-#endif
 	progname = argv[0];
 	init();
 	if (argc == 1) {	/* fake an argument list */
