--- guile-readline/readline.c.orig	2009-07-03 19:19:22.000000000 -0300
+++ guile-readline/readline.c	2010-01-09 00:20:24.000000000 -0200
@@ -435,12 +435,9 @@
 static void
 init_bouncing_parens ()
 {
-  if (strncmp (rl_get_keymap_name (rl_get_keymap ()), "vi", 2))
-    {
-      rl_bind_key (')', match_paren);
-      rl_bind_key (']', match_paren);
-      rl_bind_key ('}', match_paren);
-    }
+  rl_bind_key (')', match_paren);
+  rl_bind_key (']', match_paren);
+  rl_bind_key ('}', match_paren);
 }
 
 static int
