--- bin/varnishd/varnishd.c.orig	2009-07-25 15:39:42.000000000 -0300
+++ bin/varnishd/varnishd.c	2009-07-25 15:39:54.000000000 -0300
@@ -645,7 +645,7 @@
 	if (d_flag == 1)
 		DebugStunt();
 	if (d_flag < 2 && !F_flag)
-		AZ(inxorcise(1, d_flag));
+		AZ(daemon(1, d_flag));
 	if (d_flag == 1)
 		printf("%d\n", getpid());
 
--- bin/varnishlog/varnishlog.c.orig	2009-07-25 15:41:18.000000000 -0300
+++ bin/varnishlog/varnishlog.c	2009-07-25 15:42:15.000000000 -0300
@@ -372,7 +372,7 @@
 		exit(1);
 	}
 
-	if (D_flag && inxorcise(0, 0) == -1) {
+	if (D_flag && daemon(0, 0) == -1) {
 		perror("daemon()");
 		if (pfh != NULL)
 			vpf_remove(pfh);
--- bin/varnishncsa/varnishncsa.c.orig	2009-07-25 15:45:27.000000000 -0300
+++ bin/varnishncsa/varnishncsa.c	2009-07-25 15:46:06.000000000 -0300
@@ -569,7 +569,7 @@
 		exit(1);
 	}
 
-	if (D_flag && inxorcise(0, 0) == -1) {
+	if (D_flag && daemon(0, 0) == -1) {
 		perror("daemon()");
 		if (pfh != NULL)
 			vpf_remove(pfh);
