--- hoc.y.orig	2012-03-09 23:54:11.000000000 -0300
+++ hoc.y	2012-03-09 23:54:43.000000000 -0300
@@ -3,6 +3,7 @@
 #include "hoc.h"
 #define	code2(c1,c2)	code(c1); code(c2)
 #define	code3(c1,c2,c3)	code(c1); code(c2); code(c3)
+  extern int indef;
 %}
 %union {
 	Symbol	*sym;	/* symbol table pointer */
@@ -265,7 +266,7 @@
 	return ifno;
 }
 
-void
+int
 yyerror(char* s)	/* report compile-time error */
 {
 /*rob
@@ -309,10 +310,6 @@
 main(int argc, char* argv[])	/* hoc6 */
 {
 	static int first = 1;
-#ifdef YYDEBUG
-	extern int yydebug;
-	yydebug=3;
-#endif
 	progname = argv[0];
 	init();
 	if (argc == 1) {	/* fake an argument list */
