--- configure.orig	2008-03-16 20:09:11.000000000 -0300
+++ configure	2008-03-16 20:09:35.000000000 -0300
@@ -19076,8 +19076,7 @@
     if test -z "$arch"
     then
 	case "$host" in
-	    i386-*)           ;;
-	    i?86-*)           arch="-march=i486" ;;
+	    i?86-*)           ;;
 	    arm*-empeg-*)     arch="-march=armv4 -mtune=strongarm1100" ;;
 	    armv4*-*)         arch="-march=armv4 -mtune=strongarm" ;;
 	    powerpc-*)        ;;
